File: JavaScript/6kyu/EqualSidesOfAnArray.js
function findEvenIndex(arr) {
  var total = arr.reduce((a, b) => a + b);
  var sum = 0;
  var index = -1;

  if (total == 0) {
    return 0;
  }

  arr.map((v, i) => {
    total -= v;
    if (sum == total) {
      index = i;
    }
    sum += v;
  });

  return index;
}
// Kata

const sum = (a, from, to) => a.slice(from, to).reduce((a, b) => a + b, 0);
const findEvenIndex_kata = (a) =>
  a.findIndex((el, i) => sum(a, 0, i) === sum(a, i + 1));

var input = [1, 2, 3, 4, 3, 2, 1];
// var result = findEvenIndex_kata([1, 2, 3, 4, 3, 2, 1]);
// console.log(result);

console.log(input.slice(0, 1));
console.log(input.slice(1));

File: Python/6kyu/EqualSidesOfAnArray.py
from asyncio.windows_events import NULL
import re


def find_even_index(arr):
    m = [i for i, r in enumerate(arr) if sum(arr[:i]) == sum(arr[i+1:])]
    if not m:
        return -1

    return min(m)


arr = [1, 2, 3, 1, 0, 6]
result = find_even_index(arr)

print(arr[:result], arr[result+1:]) if result else print(result)
